packages/actor-rdf-metadata-extract-sparql-service/test/ActorRdfMetadataExtractSparqlService-test.ts

```diff
@@ -53,70 +56,78 @@ describe('ActorRdfMetadataExtractSparqlService', () => {
+    });
+
+    it('should parse relative sd:endpoint from xsd:Literal when available', async() => {
+      const input = streamifyArray([
+        DF.quad(endpointIri, serviceDescriptionEndpoint, DF.literal('/abc/../endpoint')),
+      ]);
+      await expect(actor.run({ url: endpointIri.value, metadata: input, context, requestTime })).resolves.toEqual({
+        metadata: { sparqlService: endpointService.value },
       });
     });
 
-    it('should infer HTTPS endpoint when instructed to', async() => {
+    it('should force sd:endpoint to HTTPS when instructed to', async() => {
       actor = new ActorRdfMetadataExtractSparqlService({ name: 'actor', bus, inferHttpsEndpoint: true });
-      const httpsIri = sparqlIri.value.replace(/^http:/u, 'https:');
-      const input = streamifyArray([ DF.quad(DF.namedNode(httpsIri), serviceDescriptionEndpoint, sparqlService) ]);
+      const httpsIri = endpointIri.value.replace(/^http:/u, 'https:');
+      const input = streamifyArray([
+        DF.quad(DF.namedNode(httpsIri), serviceDescriptionEndpoint, endpointService),
+      ]);
       await expect(actor.run({ url: httpsIri, metadata: input, context, requestTime })).resolves.toEqual({
-        metadata: { sparqlService: sparqlService.value.replace(/^http:/u, 'https:') },
+        metadata: { sparqlService: endpointService.value.replace(/^http:/u, 'https:') },
       });
     });
 
-    it('should parse endpoint, defaultDataset and the correct defaultGraph when available', async() => {
+    it('should parse sd:endpoint, sd:defaultDataset and the correct sd:defaultGraph when available', async() => {
       const input = streamifyArray([
-        DF.quad(sparqlIri, serviceDescriptionEndpoint, sparqlService),
-        DF.quad(sparqlIri, serviceDescriptionDefaultDataset, sparqlDefaultDataset),
-        DF.quad(sparqlDefaultDataset, serviceDescriptionDefaultGraph, sparqlDefaultGraph),
+        DF.quad(endpointIri, serviceDescriptionEndpoint, endpointService),
+        DF.quad(endpointIri, serviceDescriptionDefaultDataset, endpointDefaultDataset),
+        DF.quad(endpointDefaultDataset, serviceDescriptionDefaultGraph, endpointDefaultGraph),
         DF.quad(DF.namedNode('http://example.org/dataset2'), serviceDescriptionDefaultGraph, DF.namedNode('http://example.org/graph2')),
       ]);
-      await expect(actor.run({ url: sparqlIri.value, metadata: input, context, requestTime })).resolves.toEqual({
+      await expect(actor.run({ url: endpointIri.value, metadata: input, context, requestTime })).resolves.toEqual({
         metadata: {
-          sparqlService: sparqlService.value,
-          defaultDataset: sparqlDefaultDataset.value,
-          defaultGraph: sparqlDefaultGraph.value,
+          sparqlService: endpointService.value,
+          defaultDataset: endpointDefaultDataset.value,
+          defaultGraph: endpointDefaultGraph.value,
         },
       });
     });
-
-    it('should return empty result without endpoint defined', async() => {
```

Comment:
I guess we want to keep this one, but perhaps needs modifications?

Reply:
I am not sure what type of modifications would work. The original idea behind the test/logic was that if there is no SPARQL endpoint, then the other metadata from the Service Description does not make any sense, because it does not apply to anything. Now, it turned out parts of the metadata can apply to things outside a SPARQL endpoint (in the case of QPF). In that sense, I thought it makes sense to remove this test and modify the code so that it returns any metadata it can find, as some of it might be useful somewhere.

Reply:
Ah, I misunderstood the test then. All good!